compiler/damlc/tests/src/DA/Test/ShakeIdeClient.hs

```diff
@@ -353,7 +353,17 @@ dlintSmokeTests mbScenarioService = Tasty.testGroup "Dlint smoke tests"
             setFilesOfInterest [foo]
             expectNoErrors
             expectDiagnostic DsInfo (foo, 3, 0) "Suggestion: Use foldr"
-  ]
+    ,  testCase' "Short-circuited list comprehension" $ do
+            foo <- makeFile "Foo.daml" $ T.unlines
+                [ "daml 1.2"
+                , "module Foo where"
+                , "foo = [x | False, x <- [1..10]]" ]
+            setFilesOfInterest [foo]
+            expectNoErrors
+            expectOnlyDiagnostics []
+
```

Comment:
We should also add a test for the list literal case. Somebody might change the test above without noticing they remove a linting test.

Reply:
We'll be alright Martin. Work in progress. Smoke tests for the new linting rules in progress. Another hlint release has also been found to be necessary due to our use of the `MonadComprehensions` extension (https://github.com/ndmitchell/hlint/issues/775).